### PR TITLE
Passing key passwords through env vars and related things

### PR DESCRIPTION
**Pull request by fed239**
#29 & #30

### CODE REVIEW

File: app/server_platform.py [hunk 1]
```diff
@@ -36,33 +32,31 @@ def _params_to_pairs(params: list[Pair]) -> list[tuple[str, str]]:
...
+    hazmat_backend = default_backend()
 
     def get_ssl_context(self, tls: TLS | None) -> ssl.SSLContext:
         if tls:
-            self.update_tls_paths(tls)
             ssl_context = ssl.create_default_context()
             ssl_context.load_cert_chain(
-                tls.cert_path, tls.key_path, lambda: tls.key_password
+                os.path.join(self.OB_CERTS_DIR, tls.cert_path),
+                os.path.join(self.OB_CERTS_DIR, tls.key_path),
```
Comment (botsman) [inline app/server_platform.py:55]: I think that passing something like "../../app/broker_tls/server.pem" will allow us making calls on behalf of arbitrary certificates/keys

File: app/server_platform.py [hunk 2]
```diff
@@ -36,33 +32,31 @@ def _params_to_pairs(params: list[Pair]) -> list[tuple[str, str]]:
...
+                lambda: _read_key_password(tls.key_path)
             )
             if tls.ca_cert_path:
-                ssl_context.load_verify_locations(tls.ca_cert_path)
+                ssl_context.load_verify_locations(os.path.join(self.OB_CERTS_DIR, tls.ca_cert_path))
```
Comment (botsman) [inline app/server_platform.py:59]: same here

File: app/server_platform.py [hunk 3]
```diff
@@ -223,7 +189,11 @@ def signWithKey(
             )
 
         data = self._force_bytes(data)
-        key = self._prepare_key(open(key_path, "rb").read())
+        key = self.hazmat_backend.load_pem_private_key(
+            open(os.path.join(self.OB_CERTS_DIR, key_path), "rb").read(),
```
Comment (botsman) [inline app/server_platform.py:193]: same

File: app/server_platform.py (hunk 1 above)
Comment (fed239) [reply app/server_platform.py:55]: yes, I was thinking the same. I believe the problem (if it's a problem) was always there, but I believe it's easy to fix by checking that the paths are relative to OB_CERTS_DIR. I'll add the check

Comment (botsman) [reply app/server_platform.py:55]: I think os.path.abspath can be used together with checking that the result starts with `OB_CERTS_DIR`

**Review by botsman — APPROVED** [review]

**Pull request merged**